player/client.c

```diff
@@ -1591,35 +1594,40 @@ static bool gen_property_change_event(struct mpv_handle *ctx)
     ctx->lowest_changed = ctx->num_properties;
     for (int n = start; n < ctx->num_properties; n++) {
         struct observe_property *prop = ctx->properties[n];
-        if ((prop->changed || prop->updating) && n < ctx->lowest_changed)
+        if ((prop->changed || prop->updating || prop->updated) && n < ctx->lowest_changed)
             ctx->lowest_changed = n;
         if (prop->changed) {
             bool get_value = prop->need_new_value;
             prop->need_new_value = false;
             prop->changed = false;
-            if (prop->format && get_value) {
+            if (get_value) {
```

Comment:
You shouldn't remove the check for `prop->format`. [the initial value of `prop->need_new_value` and `prop->changed` are both set to `true`](https://github.com/mpv-player/mpv/blob/4d001bb30db6786a44658b0625dc31027767a393/player/client.c#L1466). it tries giving user a notification immediately after user observes a property. so if user observes a property with `MPV_FORMAT_NONE`([osc.lua does this](https://github.com/mpv-player/mpv/blob/4d001bb30db6786a44658b0625dc31027767a393/player/lua/osc.lua#L2284)), the `update_prop` will try retrieving a value with `MPV_FORMAT_NONE`, which causes segment fault. Changing the initial value of `need_new_value` to `format != MPV_FORMAT_NONE` should also fix it.

Reply:
Thanks for the review.
Good Catch! My bad, I somehow missed that.

I will take another look and fix it.

Reply:
just pushed a fix 1d5b202.
As a bonus we can drop the need_new_value flag completely which makes the code a bit more readable.